allauth_2fa/adapter.py

```diff
@@ -22,9 +22,13 @@ def login(self, request, user):
             request.session['allauth_2fa_user_id'] = str(user.id)
 
             redirect_url = reverse('two-factor-authenticate')
-            # Add GET parameters to the URL if they exist.
-            if request.GET:
-                redirect_url += u'?' + urlencode(request.GET)
+            # Add "next" parameter to the URL.
+            view = request.resolver_match.func.view_class()
+            view.request = request
+            success_url = view.get_success_url()
+            if success_url:
+                next_dict = {view.redirect_field_name: success_url}
+                redirect_url += u'?' + urlencode(next_dict)
```

Comment:
Hmm....I wonder if we should be including other `GET` parameters here or if that was really just to get the `next` parameter.

Reply:
The way I read the code, that was just to get the `next` parameter, without making assumptions on it actually being called `next`. With this change there's still no assumption on it actually being called `next`, since we're getting it directly from the view class that configures it. We're just also getting rid of a bunch of false-positives as well.

Reply:
Alternatively, we could do this, which preserves all existing GET params and still also works with POST:

```
            if success_url:
                query_params = request.GET.copy()
                query_params[view.redirect_field_name] = success_url
                redirect_url += u'?' + urlencode(query_params)

```

Reply:
You've convinced me :) I've updated the PR, adding tests for query param preservation and updated the code to conform to the tests.

Reply:
Just wanted to add that the solution you finally landed on is exactly how I've been doing it in my local builds. It will be great to see that in the main code base.